### ISSUE TITLE
Can't sum partially initialized Hermitian matrices that are not isbits with Hermitian Diagonal matrices

### ISSUE THREAD

**Issue by araujoms** [JuliaLang/julia#53564]
Edge case of #52895, missed by the bugfix #52942. @jishnub I think this will be easy for you.

```julia
id = Hermitian(I(2))
m = Hermitian(Matrix{BigFloat}(undef,2,2))
m.data[1,1] = 1
m.data[1,2] = 2
m.data[2,2] = 3
m+id
```

**Comment by jishnub** [comment]
The issue seems to be with summing a partially uninitialised Matrix and a Diagonal, which lacks a specialised method. I'll try to come up with a fix, which may also help with performance.

**Comment by araujoms** [comment]
I've looked into it. The problem is that when you call `UpperTriangular(id.data)` the result is a `UpperTriangular{Bool, Diagonal{Bool, Vector{Bool}}}`, that is, it is not a strided matrix, and therefore doesn't get caught by the specialized method for summing triangular matrices, but instead goes to the default sum, which cannot handle the `undef`s.

Argh, this means that whenever `Hermitian` is wrapping an abstract matrix the same problem will appear. There must be a better solution than writing a specialized method for every single type of abstract matrix. Maybe it's better to do the sum over the upper (or lower) triangular for everything, and let the sparse arrays define their own specialized methods?

**Comment by jishnub** [comment]
Yes, that makes sense. Perhaps we need to expose some function that falls back to summing over the triangular parts, and one that may be specialized by packages.
